Source/ARTRest.m

```diff
@@ -213,31 +213,27 @@ - (void)executeRequest:(NSURLRequest *)request completion:(void (^)(NSHTTPURLRes
                 stringRange = [plain rangeOfComposedCharacterSequencesForRange:stringRange];
                 NSString *shortPlain = [plain substringWithRange:stringRange];
                 // Construct artificial error
-                ARTErrorInfo *error = [ARTErrorInfo createWithCode:response.statusCode * 100 status:response.statusCode message:shortPlain];
+                error = [ARTErrorInfo createWithCode:response.statusCode * 100 status:response.statusCode message:shortPlain];
+                data = nil; // Discard data; format is unreliable.
                 [self.logger error:@"Request %@ failed with %@", request, error];
-                if (callback) {
-                    callback(response, data, error);
-                }
-                return;
             }
         }
 
         if (response.statusCode >= 400) {
-            NSError *dataError = [self->_encoders[response.MIMEType] decodeError:data];
-            if ([self shouldRenewToken:&dataError]) {
-                [self.logger debug:__FILE__ line:__LINE__ message:@"RS:%p retry request %@", self, request];
-                // Make a single attempt to reissue the token and resend the request
-                if (_tokenErrorRetries < 1) {
-                    [self executeRequest:request withAuthOption:ARTAuthenticationTokenRetry completion:callback];
-                    return;
+            if (data) {
+                NSError *dataError = [self->_encoders[response.MIMEType] decodeError:data];
+                if ([self shouldRenewToken:&dataError]) {
```

Comment:
@tcard the `shouldRenewToken` changes the `dataError` but the right now it ignores that change.